src/client.rs

```diff
@@ -1,60 +1,39 @@
 use error::{Error, ErrorObject, RequestError};
-use hyper;
-use hyper::client::RequestBuilder;
-use hyper::header::{Authorization, Basic, ContentType, Headers};
-use hyper::net::HttpsConnector;
+use hyper::{
+    self,
+    Body,
+    Request,
+    Uri,
+    HeaderMap,
+    rt::{Future, Stream},
+    client::HttpConnector,
+    header::{HeaderValue, AUTHORIZATION, CONTENT_TYPE}
+};
+use hyper_tls::HttpsConnector;
 use serde;
 use serde_json as json;
 use serde_qs as qs;
-use std::io::Read;
 
 #[derive(Clone, Default)]
 pub struct Params {
     pub stripe_account: Option<String>,
 }
 
-// TODO: #[derive(Clone)]
+#[derive(Clone)]
 pub struct Client {
-    client: hyper::Client,
+    client: hyper::Client<HttpsConnector<HttpConnector>>,
     secret_key: String,
     params: Params,
 }
 
-// TODO: With Hyper 0.11.x, hyper::Client implements clone, and we can just derive this
-impl Clone for Client {
-    fn clone(&self) -> Self {
-        let mut client = Client::new(self.secret_key.as_str());
-        client.params = self.params.clone();
-        client
-    }
-}
-
 impl Client {
     fn url(path: &str) -> String {
         format!("https://api.stripe.com/v1/{}", &path[1..])
     }
 
-    #[cfg(feature = "with-rustls")]
     pub fn new<Str: Into<String>>(secret_key: Str) -> Client {
-        use hyper_rustls::TlsClient;
-
-        let tls = TlsClient::new();
-        let connector = HttpsConnector::new(tls);
-        let client = hyper::Client::with_connector(connector);
-        Client {
-            client: client,
-            secret_key: secret_key.into(),
-            params: Params::default(),
-        }
-    }
-
-    #[cfg(feature = "with-openssl")]
-    pub fn new<Str: Into<String>>(secret_key: Str) -> Client {
-        use hyper_openssl::OpensslClient;
-
-        let tls = OpensslClient::new().unwrap();
-        let connector = HttpsConnector::new(tls);
-        let client = hyper::Client::with_connector(connector);
+        let connector = HttpsConnector::new(4).unwrap();
```

Comment:
You shouldnt be using `unwrap` like this as if something were to happen, it could cause any application using this library to panic. Maybe have the `Client` wrapped in a `Result` so it can be safely handled.

Reply:
The chances of it erroring are near to none; rust-openssl does not even compile if it can't find its needed files, and that's when it can error out when `new`ing the client. I can only see cross-compiling with non-ISO environments as a potential hazard. That's why I don't worry too much about it.